Sources/Using Presentation APIs/WindowReader.swift

```diff
@@ -0,0 +1,61 @@
+import SwiftUI
+import UIKit
+
+/// Reads the `UIWindow` that is managing the hosting of some SwiftUI content.
+struct WindowReader<Content>: View where Content: View {
+    
+    private let content: (UIWindow) -> Content
+    @State private var window: UIWindow?
+    
+    init(@ViewBuilder _ content: @escaping (UIWindow) -> Content) {
+        self.content = content
+    }
+    
+    var body: some View {
+        ZStack {
+            if let window = window {
+                content(window)
+            }
+            
+            WindowHandlerRepresentable(binding: $window)
+                .allowsHitTesting(false)
+        }
```

Comment:
> There's a layout issue with the "Selection" demo in the sample app I've not had time to look at yet, though feel free to tackle it if it's a quick win.

Seems like it's because of this. The window handler expands to take up all available space or something

Reply:
Attempt 1 to fix went terribly ;-;

![Screen Shot 2022-01-02 at 6 31 54 PM](https://user-images.githubusercontent.com/49819455/147896184-e95eae56-22b2-4eb3-bff5-cdde15de91f1.png)

Reply:
I really want to get rid of the stack, but I see that you're calling the content only when the window isn't nil. Might need to hardcode the frame then

Reply:
```swift
VStack(spacing: 0) {
    if let window = window {
        view(window)
    }
    
    WindowHandlerRepresentable(binding: $window)
        .allowsHitTesting(false)
        .frame(width: 0, height: 0)
}
```

Not ideal, but good enough for now

Reply:
Fixes the layout issue. There's still some bugs with tap-outside-to-dismiss, but everything else is looking good!

https://user-images.githubusercontent.com/49819455/147896565-15cfa0a6-718e-4cef-a3f0-55d698a6094e.mov